### PR TITLE
APS-850: new CAS1 specific report endpoint to replace existing, multiple report endpoints

### PR DESCRIPTION
**Pull request by RobBoothMOJ**
Replaced  CAS1 Report endpoints with a single report endpoint, specified in the CAS1 API spec, which takes the required report as a path parameter (enum).

Original CAS1 Report endpoints have been marked as deprecated.

### CODE REVIEW

File: src/main/resources/static/_shared.yml [hunk 1]
```diff
@@ -4720,6 +4720,14 @@ components:
           type: string
       required:
         - name
+    Cas1ReportName:
```
Comment (davidatkinsuk) [inline src/main/resources/static/_shared.yml:4723]: nice that we can enumerate the url reportName!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/ReportsController.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.reporting.properties.DailyMetricReportProperties
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.reporting.properties.LostBedReportProperties
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.reporting.properties.PlacementApplicationReportProperties
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.ReportService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserAccessService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas1.Cas1ReportService
+
+@Service("Cas1ReportsController")
+class ReportsController(
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/ReportsController.kt:24]: I know this isn't done consistently in the code base, but could we please call this 'Cas1ReportsController' to easily disambiguate it from other 'ReportsController's when searching for classes? You can then remove the name from the Service annotation

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/ReportsController.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        reportService.createLostBedReport(LostBedReportProperties(xServiceName, null, year, month), outputStream)
+      }
+      Cas1ReportName.placementApplications -> generateXlsxStreamingResponse {
+          outputStream ->
+        reportService.createCas1PlacementApplicationReport(PlacementApplicationReportProperties(year, month), outputStream)
+      }
+      Cas1ReportName.placementMatchingOutcomes -> generateXlsxStreamingResponse {
+          outputStream ->
+        cas1ReportService.createPlacementMatchingOutcomesReport(Cas1PlacementMatchingOutcomesReportProperties(year, month), outputStream)
+      }
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/ReportsController.kt:69]: this is fantastic, lots of duplication removed!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1ReportsTest.kt [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        .exchange()
+        .expectStatus()
+        .isBadRequest
+        .expectBody()
+        .jsonPath("$.detail").isEqualTo("Missing required query parameter year")
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = [0, 13])
```
Comment (davidatkinsuk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1ReportsTest.kt:34]: nice!

**Review by davidatkinsuk — COMMENTED** [review]
just one minor comment to address from me.

regarding the second commit summary - i think you are mayb burying the lede and it should be called 'add new CAS1 report endpoint' and mention in the commit description that the old one is being deprecated?

there are some other report integration tests we should probably point at the new endpoint (e.g. ApplicationReportsTest), but i can look at repointing those as part of the rewrite of the reports i'm currently working on 👍

**Review by davidatkinsuk — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/ReportsController.kt (hunk 2 above)
Comment (RobBoothMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/ReportsController.kt]: Done

**Pull request merged**